### PR TITLE
[v1] Fix try-catch for conformance tests expecting an error

### PR DESCRIPTION
**Pull request by alancai98**
## Relevant Issues
- None

## Description
- Seems like the conformance test runner was not reporting back test failures that expected an `EvaluationFail` but did not error

## Other Information
- Updated Unreleased Section in CHANGELOG: **[NO]**

- Any backward-incompatible changes? **[NO]**

- Any new external dependencies? **[NO]**

- Do your changes comply with the [Contributing Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CONTRIBUTING.md)
  and [Code Style Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CODE_STYLE.md)? **[YES]**

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by alancai98** [comment]
Some additional test failures due to this change -- https://github.com/partiql/partiql-lang-kotlin/actions/runs/10292632996.

Some of these errors are properly caught by the planner but not detected by the evaluator. Regarding the planner errors, we should probably throw on these cases -- https://github.com/partiql/partiql-lang-kotlin/issues/1543.

File: test/partiql-tests-runner/src/test/kotlin/org/partiql/runner/test/TestRunner.kt [hunk 1]
```diff
@@ -36,14 +38,16 @@ class TestRunner<T, V>(private val factory: TestExecutor.Factory<T, V>) {
                     val expect = executor.fromIon(assertion.expectedResult)
                     if (!executor.compare(actual, expect)) {
                         val ion = executor.toIon(actual)
-                        error("Expected: ${assertion.expectedResult}\nActual: $ion\nMode: ${case.compileOptions.typingMode}")
+                        throw TestRunnerException("Expected error to be thrown but none was thrown.\n${case.name}\nActual result: $ion")
```
Comment (johnedquinn) [inline test/partiql-tests-runner/src/test/kotlin/org/partiql/runner/test/TestRunner.kt:41]: This error message isn't the intention behind Assertion.EvaluationSuccess.

**Review by johnedquinn — CHANGES_REQUESTED** [review]
This PR makes me think, why are we even catching these exceptions? Using exceptions for business logic is generally advised against. IMO, this whole method can be simplified by creating the `statement`/`actual` variables in each assertion block. Then, we can use a try-catch just for the `Assertion.EvaluationFailure` where it's required.

Though, it would also be nice to have each variant of `TestCase` implement their own assertion logic.

**Comment by alancai98** [comment]
> This PR makes me think, why are we even catching these exceptions? Using exceptions for business logic is generally advised against. IMO, this whole method can be simplified by creating the `statement`/`actual` variables in each assertion block. Then, we can use a try-catch just for the `Assertion.EvaluationFailure` where it's required.
> Though, it would also be nice to have each variant of `TestCase` implement their own assertion logic.

Yeah the current `TestRunner` is somewhat confusing. This PR was initially meant to unblock some testing as part of #1542, but I'll take another look at refactoring the current `TestRunner`.

**Review by johnedquinn — APPROVED** [review]
Discussed offline -- to unblock Alan's work on the bag ops, I'm approving this PR. A follow-up will ensue.

**Pull request merged**